### PR TITLE
Add pull request template

### PR DESCRIPTION
**Pull request by superkhau**
We can update this over time as needed -- submitting my first stab at this to get the ball rolling.

@bajtos @strongloop/loopback-dev PTAL

Connect to https://github.com/strongloop/loopback/issues/2837

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<!--
+See http://loopback.io/doc/en/contrib/triaging-pull-requests.html to understand
+the MINIMUM expections for pull requests.
+-->
+
+### Bug or feature
+
+- [ ] Bug
+- [ ] Feature
```
Comment (bajtos) [inline .github/PULL_REQUEST_TEMPLATE.md]: Why is this important? Shouldn't this be obvious from the pull request description?

IMO, backwards compatibility is the most important concern to address when sending a pull request.

```
 - [ ] backwards compatible
 - [ ] breaking change(s)
```

**Comment by superkhau** [comment]
@bajtos PTAL again

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+the MINIMUM requirements for submitting pull requests.
+-->
+
+### Type of change
+
+<!--
+See http://semver.org/ for more info.
+-->
+
+- [ ] MAJOR -- Incompatible API changes (ie. breaking change(s))
```
Comment (bajtos) [inline .github/PULL_REQUEST_TEMPLATE.md]: `Backwards-incompatible changes`

1) Just "incompatible" does not say what we are not compatible with
2) It's not only about API, a change in behaviour requires SEMVER-MAJOR too.

I am also proposing to prefix all three items with `SEMVER-`, e.g. `SEMVER-MAJOR`.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+### Type of change
+
+<--
+See http://semver.org/ for more info
+-->
+
+- [ ] MAJOR -- Incompatible API changes (ie. breaking change(s))
+- [ ] MINOR -- Add functionality in a backwards-compatible manner
+- [ ] PATCH -- Backwards-compatible bug fixes
```
Comment (bajtos) [inline .github/PULL_REQUEST_TEMPLATE.md]: +1

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+
+
+### Checklist
+
+- [ ] CLA signed
+- [ ] Commit message conforms with the [Git commit message
+  guidelines](http://loopback.io/doc/en/contrib/git-commit-messages.html)
+- [ ] Code includes passing unit tests (`npm test`)
+- [ ] Code passes linting (`npm run lint` or `npm test` which also runs the
```
Comment (bajtos) [inline .github/PULL_REQUEST_TEMPLATE.md]: I would rephrase these two items:

```
- [ ] New tests are added to cover all changes
- [ ] Code passes `npm test`
```

**Comment by superkhau** [comment]
@bajtos PTAL again

**Comment by superkhau** [comment]
@0candy PTAL -- I want to land this to demo on Monday too.

cc @bajtos

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+-->
+
+### Type of change
+
+<!--
+See http://semver.org/ for more info.
+-->
+
+- [ ] SEMVER-MAJOR -- Backward-incompatible changes
+- [ ] SEMVER-MINOR -- Add functionality in a backwards-compatible manner
```
Comment (0candy) [inline .github/PULL_REQUEST_TEMPLATE.md]: Can we make it consistent? Either `backward` or `backwards`?

Comment (superkhau) [inline .github/PULL_REQUEST_TEMPLATE.md]: You're right, its a typo.

**Comment by superkhau** [comment]
@0candy PTAL again
